### PR TITLE
Allow setting only one end of the domain on QuantitativeScale

### PR DESCRIPTION
**Pull request by jtlan**
`domainMin()` sets the lower end of the domain.
`domainMax()` sets the upper end of the domain.

If only one is set, the other end of the scale will still be autodomained. If the domain would be flipped instead it is expanded using the same logic for expanding single-point domains.

If both `domainMin()` and `domainMax() are called, it's equivalent to calling `domain([min, max])`. Setting both can reverse the domain on `Scale`s that support reversal.

Calling `autoDomain()` will clear both set values, the same way calling `autoDomain()` overrides `domain()`.

Addresses #1542, although that won't be closed until this makes it to **develop**.

Demo fiddle for feature: http://jsfiddle.net/temt44gx/2/

### CODE REVIEW

File: src/scales/quantitativeScale.ts [hunk 1]
```diff
@@ -119,7 +166,29 @@ module Plottable {
...
-      return super.domain(values); // need to override type sig to enable method chaining:/
+      if (values != null) {
+        this._domainMin = values[0];
+        this._domainMax = values[1];
+      }
+      return super.domain(values);
+    }
+
+    /**
+     * Sets just the lower end of the domain.
```
Comment (bluong) [inline src/scales/quantitativeScale.ts]: I think the word "just" is unnecessary.

File: src/scales/quantitativeScale.ts [hunk 2]
```diff
@@ -119,7 +166,29 @@ module Plottable {
...
+     */
+    public domainMin(domainMin: D) {
+      this._domainMin = domainMin;
+      this._autoDomainIfAutomaticMode();
+      return this;
+    }
+
+    /**
+     * sets just the upper end of the domain.
```
Comment (bluong) [inline src/scales/quantitativeScale.ts]: Same as above.  Also, capitalize "sets" please

File: src/scales/quantitativeScale.ts [hunk 3]
```diff
@@ -25,19 +27,64 @@ module Plottable {
...
-        Utils.Methods.min<D[], D>(extents, (extent) => extent[0], defaultExtent[0]),
-        Utils.Methods.max<D[], D>(extents, (extent) => extent[1], defaultExtent[1])
-      ];
-      var includedDomain = this._includeValues(combinedExtent);
-      var paddedDomain  = this._padDomain(includedDomain);
-      return paddedDomain;
+
+      if (this._domainMin != null) {
+        extent[0] = this._domainMin;
+      }
```
Comment (bluong) [inline src/scales/quantitativeScale.ts]: Fun times with boolean operations
`extent[0] = this._domainMin || extent[0];`

File: src/scales/quantitativeScale.ts [hunk 4]
```diff
@@ -25,19 +27,64 @@ module Plottable {
...
+      if (this._domainMax != null) {
+        extent[1] = this._domainMax;
+      }
```
Comment (bluong) [inline src/scales/quantitativeScale.ts]: Same as above

**Comment by bluong** [comment]
Logic mainly looks sound.  Can we ask the requesting user to see if this performs as requested?

File: src/scales/quantitativeScale.ts (hunk 3 above)
Comment (jtlan) [inline src/scales/quantitativeScale.ts]: Right, I think the `if`-block style is more readable though.

**Comment by jtlan** [comment]
[-1] bug when `domainMin()` is set to the original max value.

File: src/scales/quantitativeScale.ts (hunk 3 above)
Comment (bluong) [inline src/scales/quantitativeScale.ts]: I personally am a fan of the suscinctness of the boolean operator though.

Comment (bluong) [inline src/scales/quantitativeScale.ts]: 0 is falsy -_-

**Pull request merged**